kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvClient.java

```diff
@@ -13,32 +10,27 @@
 import javax.enterprise.context.Dependent;
 import javax.inject.Inject;
 
-import org.apache.commons.io.IOUtils;
-import org.apache.http.HttpEntity;
-import org.apache.http.HttpResponse;
-import org.apache.http.StatusLine;
-import org.apache.http.client.methods.CloseableHttpResponse;
-import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.utils.URIBuilder;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClients;
-import org.apache.http.util.EntityUtils;
-
-import com.fasterxml.jackson.core.JsonParseException;
-import com.fasterxml.jackson.databind.JsonMappingException;
-import com.fasterxml.jackson.databind.ObjectMapper;
 
+import fi.otavanopisto.kuntaapi.server.integrations.GenericHttpCache;
+import fi.otavanopisto.kuntaapi.server.integrations.GenericHttpClient;
+import fi.otavanopisto.kuntaapi.server.integrations.GenericHttpClient.Response;
 import fi.otavanopisto.ptv.client.ApiResponse;
 import fi.otavanopisto.ptv.client.ResultType;
 
 @Dependent
 public class PtvClient extends fi.otavanopisto.ptv.client.ApiClient {
 
   private static final String BASE_PATH = "http://ptvenv.cloudapp.net:1494";
-  private static final Logger logger = Logger.getLogger(PtvClient.class.getName());
-  
+
+  @Inject
+  private Logger logger;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the "logger" field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1450)